### PR TITLE
Feature/dashboard page

### PR DESCRIPTION
**Pull request by letehaha**

### CODE REVIEW

File: src/vue/fields/DateField.vue [hunk 1]
```diff
@@ -55,12 +59,11 @@ export default {
...
-  data () {
-    return {
-      flatpickrDate: '',
-      isCalendarOpen: false
-    }
-  },
+  data: () => ({
+    flatpickrDate: '',
+    isCalendarOpen: false,
+    EVENTS
```
Comment (max-nes) [inline src/vue/fields/DateField.vue]: redundant proxying. and check all other places

File: src/vue/filters/formatDateDMY.spec.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { formatFeeSubType } from './formatFeeSubType'
```
Comment (max-nes) [inline src/vue/filters/formatDateDMY.spec.js]: remove this mess

File: src/vue/forms/CreateIssuanceForm.vue [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+          this.enableForm()
+          return
+        }
+        ErrorHandler.process(error)
+      }
+      this.enableForm()
+    },
+
+    async loadBalanceIdByEmailAndCode (email, assetCode) {
+      // TODO: add sign
```
Comment (max-nes) [inline src/vue/forms/CreateIssuanceForm.vue]: I remember you promised to check if there autosign in js-sdk before creating the PR?

File: src/vue/forms/TransferForm.vue [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+import FormConfirmation from '@/vue/common/FormConfirmation'
+
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { mapGetters, mapActions } from 'vuex'
+import { vuexTypes } from '@/vuex'
+import { base } from '@tokend/js-sdk'
+import config from '@/config'
+import { Sdk } from '@/sdk'
+import { Bus } from '@/js/helpers/event-bus'
+// FIXME: move XDR-dependent object imports to sdk
```
Comment (max-nes) [inline src/vue/forms/TransferForm.vue]: FIXME

File: src/vue/forms/TransferForm.vue [hunk 5]
```diff
@@ -0,0 +1,525 @@
...
+  FEE_TYPES
+} from '@/js/const/xdr.const'
+import { globalize } from '@/vue/filters/globalize'
+import { required, emailOrAccountId, amount } from '@validators'
+
+const VIEW_MODES = {
+  submit: 'submit',
+  confirm: 'confirm',
+  success: 'success'
+}
```
Comment (max-nes) [inline src/vue/forms/TransferForm.vue]: we don’t need `success` and `submit` modes anymore, do we?

File: src/vue/forms/TransferForm.vue [hunk 6]
```diff
@@ -0,0 +1,525 @@
...
+    FEE_TYPES,
+    ASSET_POLICIES,
+    config
+  }),
+  validations: {
+    form: {
+      amount: {
+        required,
+        amount
+        // TODO: fix it
```
Comment (max-nes) [inline src/vue/forms/TransferForm.vue]: TODO

File: src/vue/forms/TransferForm.vue [hunk 7]
```diff
@@ -0,0 +1,525 @@
+<template>
```
Comment (max-nes) [inline src/vue/forms/TransferForm.vue]: Very fat form (~500 rows). The dashboard page developing took very much time, but we still need to reduce the mess here. I’d prefer create a separate task to refactor this form

File: src/vue/pages/dashboard/Dashboard.InfoWidget.vue [hunk 8]
```diff
@@ -0,0 +1,429 @@
+<template>
```
Comment (max-nes) [inline src/vue/pages/dashboard/Dashboard.InfoWidget.vue]: looks like it copy-pasted. we can ommit this file and wait until https://github.com/tokend/web-client/pull/21 merged to reuse the tx-history template from there

File: src/vue/pages/dashboard/Dashboard.PortfolioWidget.vue [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+      />
+    </template>
+  </div>
+</template>
+
+<script>
+import config from '@/config'
+import SelectFieldCustom from '@/vue/fields/SelectFieldCustom'
+import NoDataMessage from '@/vue/common/NoDataMessage'
+// FIXME: move XDR-dependent object imports to sdk
```
Comment (max-nes) [inline src/vue/pages/dashboard/Dashboard.PortfolioWidget.vue]: FIXME

File: src/vue/pages/dashboard/Dashboard.PortfolioWidget.vue [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+            @input="$emit(EVENTS.assetChange, $event)"
+          />
+        </div>
+      </div>
+      <div class="portfolio-widget__actions">
+        <button
+          v-if="accountTypeI === ACCOUNT_TYPES.syndicate"
+          class="app__button-raised portfolio-widget__action"
+          @click="$emit(EVENTS.showCreateIssuanceForm, true)">
+          Create issuance
```
Comment (max-nes) [inline src/vue/pages/dashboard/Dashboard.PortfolioWidget.vue]: Untranslated

File: src/vue/pages/dashboard/Dashboard.PortfolioWidget.vue [hunk 11]
```diff
@@ -0,0 +1,312 @@
+<template>
```
Comment (max-nes) [inline src/vue/pages/dashboard/Dashboard.PortfolioWidget.vue]: absolute mess goes here

File: src/vue/pages/dashboard/Dashboard.vue [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import "~@scss/variables";
+@import "~@scss/mixins";
+
+$custom-breakpoint: 800px;
```
Comment (max-nes) [inline src/vue/pages/dashboard/Dashboard.vue]: unused

File: src/vue/pages/dashboard/Dashboard.vue [hunk 13]
```diff
@@ -0,0 +1,142 @@
+<template>
```
Comment (max-nes) [inline src/vue/pages/dashboard/Dashboard.vue]: the absolute mess from portfolio-widget floods this component also

File: src/validators.js [hunk 14]
```diff
@@ -1,8 +1,19 @@
 import { base } from '@tokend/js-sdk'
 import * as validators from 'vuelidate/lib/validators'
-export { minLength } from 'vuelidate/lib/validators'
 
 export const password = value => validators.minLength(6)(value)
 export const seed = value => base.Keypair.isValidSecretKey(value)
+export const amount = value => Number(value) && Number(value) > 0
+export const maxValueWrapper = value => {
```
[the remark this answers is not in the document]
Comment (letehaha) [reply src/validators.js]: wait for  PR #16

File: src/vue/directives/tableScrollShadow.js [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+    const rightShadowIsNeeded = el.scrollWidth !== el.offsetWidth
+    const leftShadowIsNeeded = el.scrollLeft > 5
+    if (rightShadowIsNeeded) {
+      el.parentNode.classList.add('app__table-right-shadow')
+    }
+    if (leftShadowIsNeeded) {
+      el.parentNode.classList.add('app__table-left-shadow')
+    }
+  },
+  unbind (el) {
```
[the remark this answers is not in the document]
Comment (letehaha) [reply src/vue/directives/tableScrollShadow.js]: `.bind()` returns not wrapper function created by itself, but my passed function. So all is fine

**Pull request merged**